File: samples.vb/Northwind/Customer.vb
Imports System
Imports System.Collections.ObjectModel
Imports System.Data.Linq
Imports System.Data.Linq.Mapping

Namespace Northwind

   <Table(Name:="Customers")>
   Public Class Customer

      <Column()>
      Public Property Address As String

      <Column()>
      Public Property City As String

      <Column()>
      Public Property CompanyName As String

      <Column()>
      Public Property ContactName As String

      <Column()>
      Public Property ContactTitle As String

      <Column()>
      Public Property Country As String

      <Association(OtherKey:="CustomerID")>
      Public Property CustomerCustomerDemos As Collection(Of CustomerCustomerDemo)

      <Column(IsPrimaryKey:=True)>
      Public Property CustomerID As String

      <Column()>
      Public Property Fax As String

      <Association(OtherKey:="CustomerID")>
      Public Property Orders As Collection(Of Order)

      <Column()>
      Public Property Phone As String

      <Column()>
      Public Property PostalCode As String

      <Column()>
      Public Property [Region] As String

      Public Sub New()
         Me.CustomerCustomerDemos = New Collection(Of CustomerCustomerDemo)
         Me.Orders = New Collection(Of Order)
      End Sub

   End Class

End Namespace

File: samples.cs/Northwind/Customer.cs
using System;
using System.Collections.Generic;
using System.Collections.ObjectModel;
using System.Data.Linq.Mapping;
using System.Linq;
using System.Text;

namespace Samples.CSharp.Northwind {

   [Table(Name = "Customers")]
   public class Customer {

      [Column(IsPrimaryKey = true)]
      public string CustomerID { get; set; }

      [Column]
      public string CompanyName { get; set; }

      [Column]
      public string ContactName { get; set; }

      [Column]
      public string ContactTitle { get; set; }

      [Column]
      public string Address { get; set; }

      [Column]
      public string City { get; set; }

      [Column]
      public string Region { get; set; }

      [Column]
      public string PostalCode { get; set; }

      [Column]
      public string Country { get; set; }

      [Column]
      public string Phone { get; set; }

      [Column]
      public string Fax { get; set; }

      [Association(OtherKey = "CustomerID")]
      public Collection<CustomerCustomerDemo> CustomerCustomerDemos { get; private set; }

      [Association(OtherKey = "CustomerID")]
      public Collection<Order> Orders { get; private set; }

      public Customer() {
         this.CustomerCustomerDemos = new Collection<CustomerCustomerDemo>();
         this.Orders = new Collection<Order>();
      }
   }
}
